2023/articles/2023-12-01.pod

```diff
@@ -0,0 +1,71 @@
+Title: Arango::Tango, the new mammal in town
+Topic: Arango::Tango
+Author: Alberto Simões <ambs@cpan.org>
+
+=encoding utf8
+
+=pod
+
+L<Arango::Tango> is not properly new, as the first versions are from 2019. Nevertheless, the number of users is still quite small as 
+other document-oriented databases, like L<ElasticSearch|https://www.elastic.co/> or C<MongoDB|https://www.mongodb.com/> are still quite
+popular. But beging somehow alergic to Java, I went in the search for an alternative and found C<ArangoDB|https://arangodb.com/>, a graph
+database (also able to deal with plain document collections) written in C++.
+
+At that time I did not find an interesting Perl module to use it, so I decided to create one. L<Arango::Tango> was born. It is, surely,
+the Perl module I own with a cuter name, with both a pun on dancing a Tango with Arango, but also with the similarity with orangutangus.
+
+ArangoDB insterface is based on REST, and the REST API is properly documented with swagger. It should be possible to use any generator
+that, from a swagger specification, generates an API. But, being REST, the API is stateless, and that does not help being proficient when
+using it. Thus, I created my own monster, that both tries to be versatile enough to allow quick implementation of new methods using a 
+structure similar to a swagger file, but also with an object-oriented approach, where each type of object we can manage in the ArangoDB database
+has a counterpart Perl object.
+
+To give a quick example on how to create a ArangoDB collection and insert a couple of documents:
+
+    use Arango::Tango;
+
+    my $server = Arango::Tango->new( host     => '127.0.0.1',
+                                     username => 'root',
+                                     password => '123123123');
+    
+    my $database = $server->create_database("database_name");
+    my $collection = $database->create_collection("collection_name");
+    
+    my $document = { name => "John", surname => "Doe" };
+    $collection->create_document( $document );
+
+    my $documents = [
+        { _key => "homer", name => "Homer", surname => "Simpson" },
+        { _key => "marge", name => "Marge", surname => "Simpson" }
+    ];
+    $collection->bulk_import( $documents );
+
+All the methods result in an HTTP request. But note that given a database and a collection objects are returned by the constructors, you
+do not need to track them yourself. You just create documents in the collection, and the module knows where the collection is (which database) and
+constructs properly the request.
+
```

Comment:
It would be interesting to see a Mojo::UserAgent or JS example (or even an interactive browser) example showing something accessing the data outside of Arango.

Reply:
I don't know if I understood correctly. I kept the article small but will try to add some examples of how to query the database.